gdc_client/upload/client.py

```diff
@@ -78,47 +73,57 @@ def upload_multipart(
     part_number,
     headers,
     verify=True,
-    pbar=None,
     ns=None,
+    debug=False,
 ):
     tries = MAX_RETRIES
     while tries > 0:
         try:
             log.debug("Start upload part {0}".format(part_number))
-            f = open(filename, "rb")
-            if OS_WINDOWS:
-                chunk_file = mmap(
-                    fileno=f.fileno(),
-                    length=num_bytes,
-                    offset=offset,
-                    access=ACCESS_READ,
-                )
-            else:
-                chunk_file = mmap(
-                    fileno=f.fileno(), length=num_bytes, offset=offset, prot=PROT_READ
+            with open(filename, "rb") as source:
+                if OS_WINDOWS:
+                    chunk_file = mmap(
+                        fileno=source.fileno(),
+                        length=num_bytes,
+                        offset=offset,
+                        access=ACCESS_READ,
+                    )
+                else:
+                    chunk_file = mmap(
+                        fileno=source.fileno(),
+                        length=num_bytes,
+                        offset=offset,
+                        prot=PROT_READ,
+                    )
+
+                log.debug("Making http request for part {}".format(part_number))
+                res = requests.put(
+                    url + "?uploadId={0}&partNumber={1}".format(upload_id, part_number),
```

Comment:
We can probably remove the numbers in these string formatters. That was a workaround to support python 2.6 on older CentOS systems, as implicit curly brace formatters were introduced in 2.7 (and py3k). I think the same CentOS systems shipped with python 3.3 as well, which supports implicit curly brace in formatted strings. Of course it doesn't hurt to keep them in.

> Changed in version 3.1: The positional argument specifiers can be omitted, so '{} {}' is equivalent to '{0} {1}'.

https://docs.python.org/3.3/library/string.html#formatstrings